perception/ImageSources.py

```diff
@@ -54,35 +54,60 @@ class VideoStream(ImageSource):
     or from the camera). Small modifications from imutils' FileVideoStream class.
     """
 
-    def __init__(self, path, transform=None, queue_size=128):
-        self.source = cv2.VideoCapture(path)
+    def __init__(self, path=None, transform=None, queue_size=128):
+        # Path not set when instantiating from camera (see from_camera class method)
+        if path is not None:
```

Comment:
What happens when the caller does not include a path? Should the code crash or does it allow ImageSources to still instantiate?

Reply:
So when you don't include a path it was meant to be used in tandem with the class method from_camera (a sort of way of overloading the constructor, per se). Anyway, this is outdated since the sequential version ended up being good enough and I got rid of all of this complexity in favor of a simple, "good enough" version